static/js/graphs_page.js

```diff
@@ -1049,19 +1054,45 @@ var graphPage = {
     },
     defaultLayoutWidget: {
         init: function (is_shared) {
-            if (utils.getURLParameter('auto_layout') || _.isNil(is_shared)) {
+            if (_.isNil(is_shared)) {
                 $('#setDefaultLayoutBtn').hide();
                 $('#removeDefaultLayoutBtn').hide();
-            } else if (utils.getURLParameter('user_layout') && utils.getURLParameter('user_layout') == default_layout_id) {
+            }else if (utils.getURLParameter('auto_layout')) {
+                $('#setDefaultLayoutBtn').show();
+                $('#removeDefaultLayoutBtn').hide();
+            }else if (utils.getURLParameter('user_layout') && utils.getURLParameter('user_layout') == default_layout_id) {
                 $('#setDefaultLayoutBtn').hide();
                 $('#removeDefaultLayoutBtn').show();
             } else {
                 $('#setDefaultLayoutBtn').show();
                 $('#removeDefaultLayoutBtn').hide();
             }
         },
-        onSetDefaultLayoutBtn: function (e) {
+        onSetDefaultAutoLayoutBtn: function (layout_id) {
```

Comment:
Change to `onSetDefaultAutoLayoutBtn` to `setDefaultAutoLayoutBtn`. `on` prefix is used for event callback functions only.